### PR TITLE
fix(market-filters): re-add Launchable

### PR DESCRIPTION
**Pull request by jaredvu**
Fix `Launchable` filter not showing up.

---

## Views

- `<MarketsDropdown>`
  - use redux state of `filter` instead of local. Now in sync with the `MarketsTable`
  - Update callback to re-set filter to `ALL` when hiding launch-able markets and on the Launchable filter
---

### CODE REVIEW

File: src/state/appUiConfigsMiddleware.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import type { PayloadAction } from '@reduxjs/toolkit';
+
+import { MarketFilters } from '@/constants/markets';
+
+import { setShouldHideLaunchableMarkets } from '@/state/appUiConfigs';
+import { setMarketFilter } from '@/state/perpetuals';
+
+import { getMarketFilter } from './perpetualsSelectors';
+
+export default (store: any) => (next: any) => async (action: PayloadAction<any>) => {
```
[the remark this answers is not in the document]
Comment (tyleroooo) [reply src/state/appUiConfigsMiddleware.ts:10]: I'm confused about how these filters work or why this is necessary too I think

[the remark this answers is not in the document]
Comment (tyleroooo) [reply src/state/appUiConfigsMiddleware.ts:10]: Can we rewrite launchable filter to just do the correct thing?

[the remark this answers is not in the document]
Comment (tyleroooo) [reply src/state/appUiConfigsMiddleware.ts:10]: ah I guess it is different

[the remark this answers is not in the document]
Comment (tyleroooo) [reply src/state/appUiConfigsMiddleware.ts:10]: yeah let's just shove it in the MarketFilter

[the remark this answers is not in the document]
Comment (jaredvu) [reply src/state/appUiConfigsMiddleware.ts:10]: kk can add the side effect to the MarketFilter

**Review by tyleroooo — APPROVED** [review]
seems good to me

**Pull request merged**